java/dataset/src/test/java/org/apache/arrow/dataset/substrait/TestAceroSubstraitConsumer.java

```diff
@@ -457,4 +465,42 @@ private static ByteBuffer getByteBuffer(String base64EncodedSubstrait) {
     substraitExpression.put(decodedSubstrait);
     return substraitExpression;
   }
+
+  @Test
+  public void testCsvConvertOptions() throws Exception {
+    final Schema schema = new Schema(Arrays.asList(
+        Field.nullable("Id", new ArrowType.Int(32, true)),
+        Field.nullable("Name", new ArrowType.Utf8()),
+        Field.nullable("Language", new ArrowType.Utf8())
+    ), null);
+    String path = "file://" + getClass().getResource("/").getPath() + "/data/student.csv";
+    BufferAllocator allocator = rootAllocator();
+    try (ArrowSchema cSchema = ArrowSchema.allocateNew(allocator);
+         CDataDictionaryProvider provider = new CDataDictionaryProvider()) {
+      Data.exportSchema(allocator, schema, provider, cSchema);
+      CsvConvertOptions convertOptions = new CsvConvertOptions(ImmutableMap.of("delimiter", ";"));
+      convertOptions.setArrowSchema(cSchema);
+      CsvFragmentScanOptions fragmentScanOptions = new CsvFragmentScanOptions(
+          convertOptions, ImmutableMap.of(), ImmutableMap.of());
+      ScanOptions options = new ScanOptions.Builder(/*batchSize*/ 32768)
+          .columns(Optional.empty())
+          .fragmentScanOptions(fragmentScanOptions)
+          .build();
+      try (
+          DatasetFactory datasetFactory = new FileSystemDatasetFactory(allocator, NativeMemoryPool.getDefault(),
+              FileFormat.CSV, path);
+          Dataset dataset = datasetFactory.finish();
+          Scanner scanner = dataset.newScan(options);
+          ArrowReader reader = scanner.scanBatches()
+      ) {
+        assertEquals(schema.getFields(), reader.getVectorSchemaRoot().getSchema().getFields());
+        int rowCount = 0;
+        while (reader.loadNextBatch()) {
+          assertEquals("[1, 2, 3]", reader.getVectorSchemaRoot().getVector("Id").toString());
```

Comment:
nit: should we check all columns?

Reply:
No. the `Id` column is enough, the test tests delimiter config and schema of `Id` change from default int64 to int32.